### ISSUE TITLE
Web transport + longpolling does not support authentication?

### ISSUE THREAD

**Issue by markerikson** [crossbario/crossbar#1205]
I'm looking to add dynamic authentication to an app I'm building, which is currently configured to use a web transport and a `longpoll` endpoint.  I'm going through the various auth examples, and trying to figure out where to add the auth options.  Digging through the code in [`checkconfig.py`](https://github.com/crossbario/crossbar/blob/274605e41d0cbe789b25583aaa37e7741e550b9c/crossbar/common/checkconfig.py), it looks like the [`check_transport_auth()`](https://github.com/crossbario/crossbar/blob/274605e41d0cbe789b25583aaa37e7741e550b9c/crossbar/common/checkconfig.py#L565) function is only called for websocket endpoints, not for longpolling.  

Based on that, it looks like it's not currently possible to add authentication to a longpoll setup.  Is there something I'm missing here?

**Comment by oberstet** [comment]
yes, the longpoll fallback isn't yet completely integrated into all the auth method machinery of crossbar.

related issues:

* https://github.com/crossbario/crossbar/issues/1077
* https://github.com/crossbario/crossbar/pull/1090
* https://github.com/crossbario/crossbar/issues/1166

**Comment by markerikson** [comment]
@oberstet : thanks for the update.  (Also, I'd forgotten that I'd commented on 1077 previously).

I did some playing around with Crossbar yesterday, and was able to successfully implement changes to the config file format and longpoll classes so that a `type : longpoll` entry accepts an `auth` field.  I'd really like to have the functionality be in an actual Crossbar release, but given that there's a couple existing issues and such, my quick-and-somewhat-dirty edits may not be the right approach for a production implementation.

I'd be happy to collaborate on getting this functionality cleaned up and merged in.

**Comment by oberstet** [comment]
@markerikson ok, cool! so if you got sth working, best to discuss / get feedback would be to make it into a PR;)

**Comment by markerikson** [comment]
@oberstet : PR filed.  The changes are pretty small - just needed to update the config definition and pass the auth settings through a couple levels so the Session can read it.
